pkg/plugin/query_repositories.go

```diff
@@ -0,0 +1,24 @@
+package plugin
+
+import (
+	"context"
+
+	"github.com/grafana/github-datasource/pkg/dfutil"
+	"github.com/grafana/github-datasource/pkg/models"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+func (s *Server) handleRepositoriesQuery(ctx context.Context, q backend.DataQuery) backend.DataResponse {
+	query := &models.RepositoriesQuery{}
+	if err := UnmarshalQuery(q.JSON, query); err != nil {
+		return *err
+	}
+	return dfutil.FrameResponseWithError(s.Datasource.HandleRepositoriesQuery(ctx, query, q))
```

Comment:
do we need to delegate to the datasource here?  why not just pull the datasource code into this file?

Reply:
The original intention is that I didn't want the datasource to be responsible for unmarshaling JSON, but the datasource and the query handlers basically have like 3-5 lines of code each, so I think you're right and they could probably be combined. I'll create a separate issue for that.